src/Equinor.ProCoSys.IPO.Command/Validators/InvitationValidators/InvitationValidator.cs

```diff
@@ -278,5 +280,29 @@ public async Task<bool> ValidSigningParticipantExistsAsync(int invitationId, int
 
             return participant.AzureOid == _currentUserProvider.GetCurrentUserOid();
         }
+
+        public async Task<bool> SameUserUnAcceptingThatAcceptedAsync(Guid invitationGuid, CancellationToken token)
+        {
+            var acceptingEvents = await (from e in _context.QuerySet<History>()
```

Comment:
Hmmm ..
Any reason for making this query as this? Why not something like this:

            var acceptingEvent = await (from e in _context.QuerySet<History>()
                join p in  _context.QuerySet<Person>() on e.CreatedById equals p.Id
                where e.ObjectGuid == invitationGuid &&
                      e.EventType == EventType.IpoAccepted
                select new
                {
                    Event = e,
                    Acceptor = p
                })
                .OrderByDescending(e => e.Event.CreatedAtUtc)
                .FirstOrDefaultAsync(token);

            return acceptingEvent != null && acceptingEvent.Acceptor.Oid == _currentUserProvider.GetCurrentUserOid();